lib/id3c/cli/command/manifest.py

```diff
@@ -73,26 +87,53 @@ def manifest():
            "Option value is parsed as a YAML fragment, so additional options supported by the sibling command \"parse-with-config\" may be inlined for testing, but you're likely better off using a config file at that point.",
     multiple = True)
 
+@click.option("--row-filter",
+    metavar = "<query>",
+    help = "The pandas query to filter rows (using the python engine) in the manifest.  "
+           "Column names refer to columns in the manifest itself.  "
+           "Example: `corrective action`.notnull() and `corrective action`.str.lower().str.startswith(\"discard\") ",
+    required = False)
+
 def parse(**kwargs):
     """
     Parse a single manifest workbook sheet.
 
-    <manifest.xlsx> must be a path or URL to an Excel workbook with at least
-    one sheet in it, identified by name using the required option --sheet.
+    <filepath> must be a path or URL to an Excel workbook or Google Sheets
+    spreadsheet with at least one sheet in it, identified by name using the required option --sheet.
     Supported URL schemes include http[s]:// and s3://, as well as others.
 
-    The required --sample-column option specifies the name of the column
-    containing the sample barcode.  Other columns may be extracted into the
-    manifest records as desired using the --extra-column option.
+    The --sample-column option specifies the name of the column
+    containing the sample barcode. The --collection-column option specifies
+    the name of the column containing the collection barcode. You must supply one
+    or both of those options.
+
+    The --date-column specifies the name of the column containing the sample collected date.
+
+    Other columns may be extracted into the manifest records as desired using the
+    --extra-column option.
+
+    The row-filter entry specifies a pandas query to filter
+    (using the python engine) rows in the manifest. Column names refer to columns
+    in the manifest itself.
+    Example: `corrective action`.notnull() and `corrective action`.str.lower().str.startswith("discard")
 
     Manifest records are output to stdout as newline-delimited JSON records.
     You will likely want to redirect stdout to a file.
     """
+    # The --sample_column and --collection_column options are not required, so
+    # check here that at least one of them is provided.
+    if kwargs.get("sample_column") is None and kwargs.get("collection_column") is None:
+        raise ValueError("You must indicate the sample column, the collection column, or both.")
```

Comment:
Raising a `click.UsageError` instead of `ValueError` here will avoid printing a stacktrace and instead print a nicer message on the command-line consistent with Click's other automatic usage checking. See an example of this in another part of ID3C:

https://github.com/seattleflu/id3c/blob/4a090fe5255205f2668f655c77f35b3ce8bfa63c/lib/id3c/cli/command/redcap_det.py#L101-L102

Reply:
That makes good sense. I ended up moving this check into _parse as you suggested below, so this went away.